nntrainer/src/elem_add_layer.cpp

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (C) 2020 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0-only
+ *
+ * @file	elem_add_layer.cpp
+ * @date	30 July 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Element-Wise Layer Class for Neural Network
+ *
+ */
+
+#include <elem_add_layer.h>
+#include <layer.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+int ElemAddLayer::initialize(bool last) {
+  int status = ML_ERROR_NONE;
+  if (input_dim.getDataLen() == 1) {
+    ml_logw("Warning: the length of previous layer dimension is one");
+  }
+
+  this->last_layer = last;
+
+  /** input dimension indicates the dimension for all the inputs to follow */
+  output_dim = input_dim;
+
+  return status;
+}
+
+sharedTensor ElemAddLayer::forwarding(sharedTensor in) {
+  input = in;
+  hidden = Tensor(in->batch(), output_dim.channel(), output_dim.height(),
+                  output_dim.width());
+  hidden.setZero();
+
+  for (unsigned int idx = 0; idx < num_inputs; ++idx) {
+    if (input_dim != in.get()[idx].getDim())
```

Comment:
I think we should either pass the input size or sharedTensor holds size sooner or later.

Reply:
Or `Layer::forwarding(sharedTensor in, int size = 1)` might work

Reply:
Then maybe have that info inside the sharedTensor itself?

Reply:
> Then maybe have that info inside the sharedTensor itself?

I like this idea better.

Reply:
> Or `Layer::forwarding(sharedTensor in, int size = 1)` might work

This won't work as the caller (NeuralNetwork) has no way to know the size while calling this function.

> Then maybe have that info inside the sharedTensor itself?

Putting size inside sharedTensor means that it should be able to represent a simple std::shared_ptr<Tensor> and std::shared_ptr<Tensor[]> with the same interface.
I will do this in a separate PR.